### PR TITLE
Bulk File Renamer

### LINKED ISSUE #1000: Bulk file renamer
# Description
I would like to create a Python based script that can rename files and directories from the command line itself. The user would be able to rename multiple files based on an ascending list of numbers, characters etc. And can also divide the files into packs of smaller folders. It would help a general user to organize his her desktop from python command line itself.

## Language

- [x] Python

## Checklist:

- **Details on the Script**

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| Bulk File Renamer and Organizer | A script to rename and organize multiple files | GSSOC21 | @devRawnie |

**Is your feature request related to a problem? Please describe.**: 

Recently I had to pick some 200 photos out of a thousand. Now they were all named using some random naming convention of the camera. It would have been so easy for me had those images be separated by date of capturing or size or any other criteria. 

**Describe the solution you'd like**: 

I want to devise a solution where the user can rename and segregate images based on different criteria such as date of creating, size, etc.

### PR DESCRIPTION
**Pull request by devRawnie**
# Description

Created python script for handling bulk file renaming and relocation.

Fixes #1000 

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by devRawnie** [comment]
Fixes issue #1000. Please look into it and merge the code

File: Python/Bulk_File_Renamer/utils.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                print(f"Deleted {len(files)} files successfully!")
+
+    def organize(self):
+        """Driver function to call organize_files method and set relevant arguments."""
+        print(f"\nFiles available to organize: {len(self.files)}")
+
+        arguments = {}
+        choice = input("Choose the method to re-organize:\n1. Rename in-place\n\
+            2. Rename and move to new folder\n3. Delete Files\n: ")
+        if choice == "1":
```
Comment (sohamsshah) [inline Python/Bulk_File_Renamer/utils.py:221]: You can replace this with switch case and put a default case for invalid choice.

**Review by sohamsshah — CHANGES_REQUESTED** [review]
Great to see you following such best practices and modularized code. I would suggest to add a README.md file to the directory which deliberates the working and use-case of the script with examples and screen shots. 

After the reverting back with suggested changes, I think its ready to be merged!

File: Python/Bulk_File_Renamer/utils.py (hunk 1 above)
Comment (devRawnie) [reply Python/Bulk_File_Renamer/utils.py:221]: Hey. Actually switch case is not a construct in Python, not that I know of. That is why I am using If-Else ladder here

Comment (devRawnie) [reply Python/Bulk_File_Renamer/utils.py:221]: Sure, I am working on the README file next

**Comment by devRawnie** [comment]
I have updated my PR with a README file

File: Python/Bulk_File_Renamer/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+3. Extension -- Extension of the files to be considered for matching
+
+### SETTINGS FOR ORGANIZING FILES
+
+1. Rename -- This setting will rename the filtered files in-place, with the name as passed in the argument. It takes 1 argument-
+   1. filename : String to be used to rename the filtered files using the format "filename_count.extension"
+2. Rename and Move -- This setting will create a New folder and copy the filtered files there after renaming them. It takes 2 arguments-
+   1. filename - String to be used to rename the filtered files using the format "filename_count.extension"
+   2. folder-name - Name to be used to create a folder, in the directory where all the files were located.
+3. Delete -- This setting will delete the filtered files.
```
Comment (sohamsshah) [inline Python/Bulk_File_Renamer/README.md:22]: Please add test case examples and screenshots. Refer the CONTRIBUTING file on how to upload images to the repo README.

Also, add a `requirements.txt` file and add commands related to that in the README.md (refer other Python Scripts on the repo to get idea)

Let me know if you need help with that.

**Review by sohamsshah — CHANGES_REQUESTED** [review]

File: Python/Bulk_File_Renamer/README.md (hunk 2 above)
Comment (devRawnie) [reply Python/Bulk_File_Renamer/README.md:22]: Ok. Will do that next.

Comment (devRawnie) [reply Python/Bulk_File_Renamer/README.md:22]: @sohamsshah  Actually my scripts do not use any external library. They use built-in Python packages. So is there any point creating a requirements.txt file?

**Comment by devRawnie** [comment]
Hi. I have updated the README.md file with the images of scripts. Kindly look into the same and merge the PR

**Review by sohamsshah — APPROVED** [review]
LGTM @HarshCasper

**Comment by devRawnie** [comment]
If the PR is finalized, @HarshCasper, Kindle merge the same

**Review by vybhav72954 — APPROVED** [review]
LGTM Merging

**Pull request merged**